### PR TITLE
Add enclave in the leadership

### PR DESCRIPTION
**Pull request by vincenthz**

### CODE REVIEW

File: src/secure/enclave.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+    pub fn from_vec(leaders: Vec<Leader>) -> Self {
+        let mut e = Self::new();
+        for leader in leaders {
+            e.add_leader(leader);
+        }
+        e
+    }
+
+    pub fn get_leaderids(&self) -> Vec<LeaderId> {
```
Comment (mzabaluev) [inline src/secure/enclave.rs]: Nitpick: name `get_leader_ids` for snake case consistency.

File: src/secure/enclave.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    pub fn new() -> Self {
+        Enclave {
+            leaders: Arc::new(RwLock::new(BTreeMap::new())),
+        }
+    }
+
+    pub fn from_vec(leaders: Vec<Leader>) -> Self {
+        let mut e = Self::new();
+        for leader in leaders {
+            e.add_leader(leader);
```
Comment (mzabaluev) [inline src/secure/enclave.rs]: It's better to construct the whole map and then put it under the lock. Shouldn't be much of a difference with a small number of leaders, though.

File: src/secure/enclave.rs (hunk 1 above)
Comment (vincenthz) [reply src/secure/enclave.rs]: this will go away once this is used as an abstract box

File: src/secure/enclave.rs (hunk 2 above)
Comment (vincenthz) [reply src/secure/enclave.rs]: I'ld agree but it's also to load 0 to 2 leaders probably, and in very soon future will be done by a control-api REST endpoint

**Pull request merged**